scripts/rmc.py

```diff
@@ -93,19 +93,86 @@ def symbol_table_to_gotoc(json_filename, cbmc_filename, verbose=False, keep_temp
     return process.returncode
 
 
-def run_cbmc(cbmc_filename, cbmc_args, verbose=False, quiet=False):
+def run_cbmc(cbmc_filename, cbmc_args, verbose=False, quiet=False, output_to=None):
     cbmc_cmd = ["cbmc"] + cbmc_args + [cbmc_filename]
     process = subprocess.Popen(cbmc_cmd, universal_newlines=True,
                                stdout=subprocess.PIPE,
                                stderr=subprocess.STDOUT)
-    if not quiet:
+    if output_to != None:
+        with open(output_to, "w") as f:
+            for line in iter(process.stdout.readline, ""):
+                f.write(line)
+    elif not quiet:
         for line in iter(process.stdout.readline, ""):
             print(line, end="")
     process.stdout.close()
     retcode = process.wait()
     print_rmc_step_status("cbmc", process, verbose)
     return retcode
 
+def run_visualize(cbmc_filename, cbmc_args, verbose=False, quiet=False, keep_temps=False):
+    results_filename = "results.xml"
+    coverage_filename = "coverage.xml"
+    property_filename = "property.xml"
+    temp_goto_filename = "temp.goto"
+    if not keep_temps:
+        for filename in [results_filename, coverage_filename, property_filename, temp_goto_filename]:
+            atexit.register(delete_file, filename)
```

Comment:
I've never seen atexit before, but I'm not sure what keep_temps has to do with deleting files before overrwriting them.

Reply:
I got this from another place in the code. The atexit.register command applies the given function to the given value before the full program terminates. It's used to clean-up temporary files, but `keep_temps` tells us not to do this. It doesn't delete before overwriting.